proto/cmp/services/transport/v3/trip_types.proto

```diff
@@ -0,0 +1,151 @@
+import "cmp/types/v2/location.proto";
+import "cmp/types/v2/price.proto";
+import "cmp/types/v2/product_code.proto";
+import "cmp/types/v2/service_fact.proto";
+import "google/protobuf/timestamp.proto";
+
+// This message type represents a one way trip, either travelling or returning.
+//
+// ![Diagram](https://storage.googleapis.com/docs-cmp-files/diagrams/proto/cmp/services/transport/v3/trip_types.proto.dot.xs.svg)
+//
+// [Open Message Diagram](https://storage.googleapis.com/docs-cmp-files/diagrams/proto/cmp/services/transport/v3/trip_types.proto.dot.svg)
+message Trip {
+  // Schedule for the whole trip
+  cmp.services.transport.v3.TripSchedule schedule = 1;
+
+  // Luggage
+  //
+  // Ex: `3`
+  cmp.types.v1.Baggage baggage = 2;
+
+  // Price
+  //
+  // Ex: `Price(net=240, currency=Currency.CURRENCY_EUR)
+  cmp.types.v2.Price price = 3;
+
+  // Departure and Arrival dates and times can be derived from the first and last segments
+  // Trip segments are the legs offered
+  repeated cmp.services.transport.v3.TripSegment segments = 4;
+}
+
+// Represents a departure or arrival event
+message TransitEvent {
+  // Date time of the event
+  google.protobuf.Timestamp date_time = 1;
+
+  // Event location
+  cmp.types.v2.LocationCode location_code = 2;
+}
+
+// A segment of a trip
+message TripSegment {
+  // Segment Info
+  cmp.services.transport.v3.SegmentInfo info = 1;
+
+  // Flights, trains and transfers often offer different service types
+  // for a product. Think about economy or business for flights, first and second
+  // class on trains and a shared shuttle or a private transfer.
+  //
+  // Service type code examples:
+  // flight: "Y", "J", "F".
+  // train: "1st", "2nd" or "S", "C", "P"
+  // transfer: "S", "P", "VIP", "SS"
+  string service_type_code = 2;
+
+  // Service type description examples:
+  // flight: "Economy Class", "Business Class", "First Class"
+  // train: "First Class", "Second Class" or "standard", "comfort", and "premium",..
+  // transfer: "Shuttle", "Private", "VIP Limosine", "Speedy Shuttle",...
+  string service_type_description = 3;
+
+  // Included, optional or compulsary services for this segment or leg. Just a code and
+  // description can be included if it is a package (seat, priority, large carry-on).
+  // or a price detail can be included for optionally pre-ordered meals for example.
+  repeated cmp.types.v2.ServiceFact services = 4;
+
+  // Min PAX
+  //
+  // Ex: `1`
+  int32 min_pax = 5;
+
+  // Max PAX
+  //
+  // Ex: `3`
+  int32 max_pax = 6;
+}
+
+// A whole trip schedule from the first segment till the last one
+message TripSchedule {
+  // Carrier Code
+  string carrier_code = 1;
+
+  // Carrier Type. Ex: airline, transfer company, train company, .. etc
+  string carrier_type = 2;
+
+  // Start and end datetime for the trip
+  cmp.types.v1.DateTimeRange period = 3;
+
+  // Schedule for each segment of the trip
+  repeated cmp.services.transport.v3.SegmentInfo segments = 4;
```

Comment:
We have the segments in the `Trip` type and also here. Why is that? Isn't that means we have the same info in two different places in the same service?

Reply:
Oh, this is `SegmentInfo` type, not `TripSegment`. This is confusing, why we have segment info in the trip schedule type?